modules/tm/uac.c

```diff
@@ -553,6 +553,19 @@ int t_uac(str* method, str* headers, str* body, dlg_t* dialog,
 		request->buffer.len = buf_len;
 	}
 
+	if (ref_script_route_is_valid(tm_local_request)) {
+		LM_DBG("Found Local-Request Route...\n");
+		memset(&dummy_msg, 0, sizeof(struct sip_msg));
+		dummy_msg.buf = request->buffer.s;
+		dummy_msg.len = request->buffer.len;
+
+		if (parse_msg(request->buffer.s, request->buffer.len, &dummy_msg) == 0) {
+			LM_DBG("Parsed Message, executing Local-Reply Route with Message...\n");
+			run_top_route(sroutes->request[tm_local_request->idx], &dummy_msg);
```

Comment:
when comes to this new local_request, I see it is triggered right after the existing local_route. I see nothing in between to justify one more route. You mentioned some callbacks (being executed between the triggering of the old and new local request route), but I fail to  see any.....

Reply:
I need to double-check, but I added it since the request passed to the locally executed route did not include all the changes applied to the message (like content changed from the rtp_relay module). This is why I explicitly parsed the newly created buffer instead of using the other route.

Reply:
OK, I think I understand what is happening here. When the `local_route` is executed, the buffer (with all the changes so far) is constructed, but BEFORE re-parsing (to get a `sip_msg`), the TMCB_LOCAL_REQUEST_OUT callbacks are executed. So the the changes generated by TMCB_LOCAL_REQUEST_OUT are not seen in `local_route`.. and rtp_relay uses this TMCB_LOCAL_REQUEST_OUT callback.. I will check to see if we can change the order and fix this.